src/esp/CMakeLists.txt

```
@@ -17,15 +17,595 @@ if(BUILD_WARNINGS_AS_ERRORS)
+  core/Esp.h
+  core/Logging.cpp
+  core/Logging.h
+  core/managedContainers/AbstractFileBasedManagedObject.h
+  core/managedContainers/AbstractManagedObject.h
+  core/managedContainers/ManagedContainer.h
+  core/managedContainers/ManagedContainerBase.cpp
+  core/managedContainers/ManagedContainerBase.h
+  core/managedContainers/ManagedFileBasedContainer.h
+  core/Random.h
+  core/Spimpl.h
+  core/Utility.h
+)
+
+#################
+## geo
+
+set(
+  geo_SOURCES
+  geo/CoordinateFrame.cpp
+  geo/CoordinateFrame.h
+  geo/Geo.cpp
+  geo/Geo.h
+  geo/OBB.cpp
+  geo/OBB.h
+)
+
+#################
+## gfx
+set(
+  gfx_SOURCES
+  gfx/CubeMap.cpp
+  gfx/CubeMap.h
+  gfx/Drawable.cpp
+  gfx/Drawable.h
+  gfx/DrawableConfiguration.cpp
+  gfx/DrawableConfiguration.h
+  gfx/DrawableGroup.cpp
+  gfx/DrawableGroup.h
+  gfx/GenericDrawable.cpp
+  gfx/GenericDrawable.h
+  ${gfx_SOURCES}
+  gfx/SkinData.h
+  gfx/MeshVisualizerDrawable.cpp
+  gfx/MeshVisualizerDrawable.h
+  gfx/LightSetup.cpp
+  gfx/LightSetup.h
+  gfx/magnum.h
+  gfx/RenderCamera.cpp
+  gfx/RenderCamera.h
+  gfx/CubeMapCamera.cpp
+  gfx/CubeMapCamera.h
+  gfx/CubeMap.cpp
+  gfx/CubeMap.h
+  gfx/DebugLineRender.cpp
+  gfx/DebugLineRender.h
+  gfx/Renderer.cpp
+  gfx/Renderer.h
+  gfx/replay/Keyframe.h
+  gfx/replay/Player.cpp
+  gfx/replay/Player.h
+  gfx/replay/Recorder.cpp
+  gfx/replay/Recorder.h
+  gfx/replay/ReplayManager.h
+  gfx/replay/ReplayManager.cpp
+  gfx/WindowlessContext.cpp
+  gfx/WindowlessContext.h
+  gfx/RenderTarget.cpp
+  gfx/RenderTarget.h
+  gfx/ShaderManager.cpp
+  gfx/ShaderManager.h
+  gfx/PbrShader.cpp
+  gfx/PbrShader.h
+  gfx/PbrDrawable.cpp
+  gfx/PbrDrawable.h
+  gfx/TextureVisualizerShader.cpp
+  gfx/TextureVisualizerShader.h
+  gfx/CubeMapShaderBase.cpp
+  gfx/CubeMapShaderBase.h
+  gfx/DoubleSphereCameraShader.cpp
+  gfx/DoubleSphereCameraShader.h
+  gfx/EquirectangularShader.cpp
+  gfx/EquirectangularShader.h
+  gfx/PbrIBLHelper.cpp
+  gfx/PbrIBLHelper.h
+  gfx/PbrEquiRectangularToCubeMapShader.cpp
+  gfx/PbrEquiRectangularToCubeMapShader.h
+  gfx/PbrPrecomputedMapShader.cpp
+  gfx/PbrPrecomputedMapShader.h
+  gfx/PbrTextureUnit.h
+  gfx/GaussianFilterShader.h
+  gfx/GaussianFilterShader.cpp
+)
+
+if(BUILD_WITH_BACKGROUND_RENDERER)
+  list(APPEND gfx_SOURCES gfx/BackgroundRenderer.h gfx/BackgroundRenderer.cpp)
+endif()
+
+list(APPEND gfx_SOURCES ${GfxShaderResources})
+
+#################
+## assets
+set(
+  assets_SOURCES
+  assets/Asset.cpp
+  assets/Asset.h
+  assets/BaseMesh.cpp
+  assets/BaseMesh.h
+  assets/CollisionMeshData.h
+  assets/GenericSemanticMeshData.cpp
+  assets/GenericSemanticMeshData.h
+  assets/GenericMeshData.cpp
+  assets/GenericMeshData.h
+  assets/MeshData.h
+  assets/MeshMetaData.h
+  assets/RenderAssetInstanceCreationInfo.cpp
+  assets/RenderAssetInstanceCreationInfo.h
+  assets/ResourceManager.cpp
+  assets/ResourceManager.h
+  assets/RigManager.cpp
+  assets/RigManager.h
+)
+
+list(APPEND assets_SOURCES ${PbrIBlImageResources})
```

```suggestion
  assets/RigManager.h
  ${PbrIBlImageResources}
)
```

feels cleaner to me, same with other resources.